### PR TITLE
Switch to a declarative interface for table definitions

### PR DESCRIPTION
**Pull request by lukasjuhrich**
@ibot3 please check that all tables you touched/introduced in your PR still work as desired.

### CODE REVIEW

File: web/blueprints/helpers/table.py [hunk 1]
```diff
@@ -258,3 +342,30 @@ def datetime_format(dt, default=None):
...
+    url_parts[4] = urlencode(query_parts)
+    return urlunparse(url_parts)
+
+
+@lazy_join
+def button_toolbar(title: str, href: str, icon: str = "glyphicon-plus")\
+        -> LazilyJoined:
+    params = html_params(class_="btn btn-primary", href=href)
+    yield f"<a {params}>"
+    yield f"<span class=\"glyphicon {icon}\"></span>"
```
Comment (ibot3) [inline web/blueprints/helpers/table.py]: There should be a space after the icon.

File: web/blueprints/infrastructure/tables.py [hunk 2]
```diff
@@ -1,77 +1,55 @@
...
-            Column('edit_link', 'Editieren', formatter='table.btnFormatter', width=1),
-            Column('delete_link', 'Löschen', formatter='table.btnFormatter', width=1)
-        ], **kw)
-
-    def generate_toolbar(self):
+    id = Column("#")
+    name = LinkColumn("Name")
+    ip = Column("Management IP")
+    edit_link = Column('Editieren', formatter='table.btnFormatter', width=1)
+    delete_link = Column('Löschen', formatter='table.btnFormatter', width=1)
```
Comment (ibot3) [inline web/blueprints/infrastructure/tables.py]: There should be used a `BtnColumn` here too.

File: web/blueprints/facilities/tables.py [hunk 3]
```diff
@@ -1,77 +1,58 @@
...
+    class Meta:
+        table_args = {'data-sort-name': 'room'}
 
 
 class PatchPortTable(BootstrapTable):
+    name = Column('Name', width=2)
+    room = Column('→ Raum', formatter='table.linkFormatter', width=5)
+    switch_port = Column('→ Switch-Port', formatter='table.linkFormatter', width=3)
+    edit_link = Column('Editieren', formatter='table.btnFormatter')
+    delete_link = Column('Löschen', formatter='table.btnFormatter')
```
Comment (ibot3) [inline web/blueprints/facilities/tables.py]: There should be used a `BtnColumn` here too.

File: web/blueprints/user/tables.py [hunk 4]
```diff
@@ -46,92 +39,76 @@ class MembershipTable(BootstrapTable):
     In the toolbar, a “new membership” button is inserted if the
     :py:obj:`current_user` has the ``add_membership`` property.
     """
+    group_name = Column("Gruppe")
+    begins_at = Column("Beginn", formatter='table.dateFormatter')
+    ends_at = Column("Ende", formatter='table.dateFormatter')
+    actions = Column("Aktionen", formatter='table.multiBtnFormatter')
```
Comment (ibot3) [inline web/blueprints/user/tables.py]: A `MultiBtnColumn` and `DateColumn` could be worth to think about.

File: web/blueprints/helpers/table.py (hunk 1 above)
Comment (lukasjuhrich) [reply web/blueprints/helpers/table.py]: What do you mean „after the icon“? After the `</span>`, or inbetween, or after `{icon}`?

Comment (lukasjuhrich) [reply web/blueprints/helpers/table.py]: Resolved in f7b046d8a4c51784ea5618c31427cbcd0d14d0c5

File: web/blueprints/facilities/tables.py (hunk 3 above)
Comment (lukasjuhrich) [reply web/blueprints/facilities/tables.py]: Resolved in 6b41f59866d27292d84120ba8391cb042618f21f

File: web/blueprints/user/tables.py (hunk 4 above)
Comment (lukasjuhrich) [reply web/blueprints/user/tables.py]: Resolved in 5689730222cbb88cada268b21b2d65ce5730afc9, d75faddd19c5ae967bf2344117d55ee7d0b34655

File: web/blueprints/infrastructure/tables.py (hunk 2 above)
Comment (lukasjuhrich) [reply web/blueprints/infrastructure/tables.py]: Resolved in 6b41f59866d27292d84120ba8391cb042618f21f

**Pull request merged**